app/forms/TaxResult.scala

```diff
@@ -59,7 +62,7 @@ object TaxResult {
   ): CalculatorResponse =
     new Calculator(
       extractTaxCode(quickCalcAggregateInput, defaultTaxCodeProvider),
-      true,
+      if (quickCalcAggregateInput.savedTaxCode.exists(_.gaveUsTaxCode)) true else false,
```

Comment:
Don't think you need the if statement here as `quickCalcAggregateInput.savedTaxCode.exists(_.gaveUsTaxCode)` will give you the boolean result anyway.